voting/src/voting.rs

```diff
@@ -101,6 +114,45 @@ impl Context {
                 self.storage_mut().set(vote_paper.vote_paper_id.0.as_ref(), serde_cbor::to_vec(&vote_paper).unwrap());
                 Ok(vote_paper.vote_paper_id)
             }
+            VOTE_TX_TYPE => {
+                let tx: VoteTransaction =
+                    serde_cbor::from_slice(&transaction.body()).map_err(|_| ExecuteError::InvalidFormat)?;
+                let vote_paper_id = tx.action.vote_paper_id;
+                let choice = tx.action.choice;
+                let voter_signature = tx.action.voter_signature;
+
+                if choice != VoteChoice::Favor && choice != VoteChoice::Against && choice != VoteChoice::Absention {
+                    return Err(ExecuteError::InvalidChoice)
+                }
+
+                if !self.storage().has(vote_paper_id.0.as_ref()) {
+                    return Err(ExecuteError::VotePaperNotFound)
+                }
+                let mut vote_paper: VotePaper = {
+                    let bytes = self
+                        .storage()
+                        .get(vote_paper_id.0.as_ref())
+                        .expect("Previously we checked the existance of the vote paper in the state");
+                    serde_cbor::from_slice(&bytes).expect("Vote paper is serialized by this code")
+                };
+
+                if !vote_paper.verify_signature(&voter_signature) {
+                    return Err(ExecuteError::InvalidVoterSignature)
+                }
+
+                let vote = Vote::new(vote_paper_id, choice, voter_signature);
+                let vote_id = vote.vote_id.clone();
+                vote_paper.set_used_in(vote_id.clone()).map_err(|_| ExecuteError::UsedVotePaper)?;
+                self.storage_mut().set(vote_id.as_ref(), serde_cbor::to_vec(&vote).unwrap());
+
+                let meeting_id = vote_paper.get_general_meeting_id();
+
+                let box_key = crate::generate_voting_box_key(&meeting_id);
+                let mut vote_box = self.meeting_mut().get_vote_box(box_key.as_slice()).unwrap();
+                vote_box.drop_in_box(vote.vote_id.clone());
+
```

Comment:
How about moving this code to "Add a missing function to update vote box changes" commit?